command/node/drain.go

```diff
@@ -54,14 +69,127 @@ func Drain(c *cli.Context, logger *log.Logger) error {
 	}
 
 	if len(matches) == 0 {
-		return fmt.Errorf("Could not find any nodes matching provided filters")
+		return fmt.Errorf("could not find any nodes matching provided filters")
 	}
 
 	var wg sync.WaitGroup
 	ctx := context.Background()
 
 	for _, node := range matches {
 		log.Infof("Node %s (class: %s / version: %s)", node.Name, node.NodeClass, node.Attributes["nomad.version"])
+		if c.Bool("with-benefits") {
+			log.Infof("Drain mode with benefits selected, marking node as ineligible and starting to move the jobs to the specified constraint")
+			_, err := nomadClient.Nodes().ToggleEligibility(node.ID, false, nil)
+			if err != nil {
+				log.Errorf("Error updating scheduling eligibility for %s: %s", node.Name, err)
+				continue
+			}
+			// Bring the allocations running on the node
+			nodeAllocations, _, err := nomadClient.Nodes().Allocations(node.ID, nil)
+			if err != nil {
+				log.Errorf("Error updating scheduling eligibility for %s: %s", node.Name, err)
+				continue
+			}
+
+			for _, nodeAllocation := range nodeAllocations {
+				if *nodeAllocation.Job.Type != nomadStructs.JobTypeService {
```

Comment:
wondering if we also want to include batch jobs too or just to make it a flag to include all types of jobs, else default to service only

Reply:
according to docs:

```
The node drain command is used to toggle drain mode on a given node. Drain mode prevents any new tasks from being allocated to the node, and begins migrating all existing allocations away. Allocations will be migrated according to their migrate stanza until the drain's deadline is reached.
```

and 
```
The migrate stanza specifies the group's strategy for migrating allocations from draining nodes. If omitted, a default migration strategy is applied. If specified at the job level, the configuration will apply to all groups within the job. Only service jobs with a count greater than 1 support migrate stanzas.
```

so other types of jobs are not implacable to benefits ;)

Reply:
good to know!